hoomd/mesh.py

```diff
@@ -63,9 +62,7 @@ class Mesh(_HOOMDBaseObject):
     def __init__(self):
 
         param_dict = ParameterDict(size=int,
-                                   types=OnlyIf(
-                                       to_type_converter([str]),
-                                       preprocess=self._preprocess_type))
+                                   types=OnlyIf(to_type_converter([str])))
```

Comment:
```python
types=[str])
```

Reply:
About not erroring when we set `type_ids`: I don't error it so the user can add a suitable triangulation afterwards. Imagine yu want to delete triangles from your mesh. In that case you have to first set the new triangulation ids without these specific to the removed triangles. Afterwards the user passes the updated triangulation. If I introduce an error, step 1 wouldn't work. I could add a warning though.  

It is also not in `ParameterDict` as the array both dictates the ids of the triangles and edges in the mesh.

Regarding exposing the attribute. I think this is still the best way to avoid to store a `TriangleData::Snapshot`.